below/procfs/src/lib.rs

```diff
@@ -1305,21 +1309,35 @@ impl NetReader {
     }
 
     pub fn read_netstat(&self) -> Result<NetStat> {
-        let netstat_map = self.read_kv_diff_line("netstat")?;
-        let snmp_map = self.read_kv_diff_line("snmp")?;
-        let snmp6_map = self.read_kv_same_line("snmp6")?;
+        let netstat_map = handle_io_error(&self.logger, self.read_kv_diff_line("netstat"))?;
+        let snmp_map = handle_io_error(&self.logger, self.read_kv_diff_line("snmp"))?;
+        let snmp6_map = handle_io_error(&self.logger, self.read_kv_same_line("snmp6"))?;
+        let iface_map = handle_io_error(&self.logger, self.read_net_map())?;
```

Comment:
Just calling out that these files could (unlikely though) be missing. So I think handling ENOENT here is reasonable. For example ipv4 files could be missing with CONFIG_INET=n. Or if an iface gets removed while we're reading it's directory.